### PR TITLE
testament: catch failing nimble tests and clean up a bit

### PR DESCRIPTION
**Pull request by narimiran**

### CODE REVIEW

**Comment by Araq** [comment]
On travis c2nim and nimx now fail, yay.

File: testament/categories.nim [hunk 1]
```diff
@@ -475,47 +467,42 @@ proc getPackageDir(package: string): string =
...
-        of pfExtraOnly:
-          for n, cmd, commit, hasDeps in important_packages.packages.items:
-            if name == n:
-              let cmd = if cmd.len == 0: defaultCmd else: cmd
-              yield (name, url, cmd, hasDeps)
-        of pfAll:
-          yield (name, url, defaultCmd, false)
+        let cmd = if cmd.len == 0: defaultCmd else: cmd
+        yield (name, url, cmd, hasDeps)
+        break
```
Comment (Araq) [inline testament/categories.nim]: I think a ``yield + break`` turns this iterator into a regular proc?

Comment (narimiran) [reply testament/categories.nim]: Not really.

`break` just takes you out of the inner `for` loop (where you search the whole nimble package list for your package `n` from the outer `for` loop).
After the `break`, you search for the next package.

Comment (Araq) [reply testament/categories.nim]: Ok.

**Pull request closed without merging**